Remove histogramdd functional wrapper

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74201

Merge once the forward compatibility period is expired for the histogramdd
operator.

peterbell10 committed:
Remove histogramdd functional wrapper

Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Fix int-list overloading with TensorList on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 commented:
The forward compatibility period is now expired.

peterbell10 committed:
Fix flake8 on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

ezyang reviewed (approved):
hurray

ezyang commented:
@pytorchbot merge this

peterbell10 committed:
Check int-list tensors are scalars during tracing on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 commented on torch/csrc/utils/python_arg_parser.cpp:
```diff
@@ -546,11 +546,15 @@ static bool is_int_list(PyObject* obj, int broadcast_size) {
-      (jit::tracer::isTracing() && THPVariable_Check(item.ptr())));
+        jit::tracer::isTracing() &&
+        THPVariable_Check(item.ptr()) &&
+        THPVariable_Unpack(item.ptr()).sizes() == c10::IntArrayRef{});
```
The slow test that was failing was running tracing on the `TensorList` overload, but because of this check it was actually calling the `IntList` overload and failing.  This fixes the ambiguity in this particular test. A better solution would be if Tensors representing ints being traced were marked with a bitfield similar to `is_wrapped_number_` for wrapped scalar values. This works for now though, since you would never want to pass a scalar tensor as the `bins` argument for `histogramdd` and it seems to be the only function running into this issue.

ezyang commented:
@pytorchbot merge this

suo commented:
@pytorchbot revert this

Looks like this raced with @anjali411's change https://hud.pytorch.org/pytorch/pytorch/commit/8cc338e5c2e3f1b6512dc3b33d33281ac5f4357c

peterbell10 committed:
Update on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

peterbell10 commented on test/allowlist_for_publicAPI.json:
```diff
@@ -4015,6 +4015,7 @@
     "hinge_embedding_loss",
     "histc",
     "histogram",
+    "histogramdd",
```
Everything C++ bound function seems to have no `__module__` set.

peterbell10 committed:
Set histogramdd.__module__ on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

albanD commented on torch/_torch_docs.py:
```diff
@@ -4526,6 +4526,97 @@ def merge_dicts(*dicts):
+                      tensor([0.0000, 0.5000, 1.0000])))
+
+""")
+torch.histogramdd.__module__ = "torch"
```
Thanks for the small fix, could you add `TODO: Fix via https://github.com/pytorch/pytorch/issues/75798` just to make sure we don't loose track of that one-time fix?

peterbell10 replied:
Done

peterbell10 committed:
Add TODO on "Remove histogramdd functional wrapper"


Merge once the forward compatibility period is expired for the histogramdd
operator.

[ghstack-poisoned]

albanD reviewed (approved):
Thanks!
Waiting for CI then you can land

peterbell10 commented:
@pytorchbot merge this
